File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int l = 0;
        int longest = 0;
        set<char> eliminate;

        for(int r=0;r<s.length();r++) {
            while(eliminate.find(s[r]) != eliminate.end()) {
                eliminate.erase(s[l]);
                l++;
            }
            eliminate.insert(s[r]);
            longest = max(longest, r-l+1);
        }
        return longest;
    }
};

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        n = len(s)
        unique = set()
        l,r = 0,0
        max_len = 0

        while r < n:
            while s[r] in unique:
                unique.remove(s[l])
                l += 1
            unique.add(s[r])
            max_len = max(max_len, r - l + 1)
            r += 1
        return max_len